### PR TITLE
Reactor changes

### PR DESCRIPTION
**Pull request by BeagleGaming1**
# About the pull request

Updates reactors, fuel cells, fuel cell recyclers, and a proc that got in the way
plus reactor sprites attempt 2

Reactor sprites originally from Cosmic-Overlord/TerraGov-Marine-Corps#227, modified by me

# Explain why it's good for the game

Adding more complexity and content to reactors makes them more than just a set piece that gets set up and ignored

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

I didnt test it
</details>

# Changelog
:cl:
balance: Xenos can slash reactors
balance: Reactors that are damaged will still run until reaching weld damage or being manually de-activated
balance: Reactors that are damaged will lose fuel (when applicable) and fail more often when fuel runs out, new cells need to be used to reset the fail rate
balance: Reactors can be damaged by bullets or explosions
code: Updated Power all ship reactors admin button code
imageadd: Updated reactor sprites
maptweak: Minor changes to reactors and surrounding areas
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by nauticall — APPROVED** [review]
ayep

File: code/game/machinery/fuelcell_recycler.dm [hunk 1]
```diff
@@ -1,144 +1,217 @@
...
+
+/obj/structure/machinery/fuelcell_recycler/proc/recharge_cell(obj/item/fuel_cell/cell)
+	cell.modify_fuel(recharge_amount)
+	if(!cell.new_cell)
+		cell.new_cell = TRUE
+
+/obj/structure/machinery/fuelcell_recycler/full/Initialize(mapload, ...)
+	. = ..()
+	cell_left = new
+	cell_right = new
```
Comment (harryob) [inline code/game/machinery/fuelcell_recycler.dm:173]:
```suggestion
	cell_left = new(src)
	cell_right = new(src)
```
shouldn't these be in the recyclers contents?

File: code/game/machinery/fuelcell_recycler.dm [hunk 2]
```diff
@@ -1,144 +1,217 @@
...
+/obj/item/fuel_cell/get_examine_text(mob/user)
+	.  = ..()
+	if(ishuman(user))
+		. += "The fuel indicator reads: [get_fuel_percent()]%"
+
+/obj/item/fuel_cell/proc/get_fuel_percent()
+	return round(100*fuel_amount/max_fuel_amount)
+
+/obj/item/fuel_cell/proc/is_regenerated()
+	return(fuel_amount == max_fuel_amount)
```
Comment (harryob) [inline code/game/machinery/fuelcell_recycler.dm:211]:
```suggestion
	return (fuel_amount == max_fuel_amount)
```

File: code/game/machinery/fuelcell_recycler.dm [hunk 3]
```diff
@@ -1,144 +1,217 @@
...
+
+/obj/item/fuel_cell/proc/modify_fuel(amount) //increase or decrease fuel as needed
```
Comment (harryob) [inline code/game/machinery/fuelcell_recycler.dm:213]: not a dmdoc?

File: code/game/machinery/fuelcell_recycler.dm [hunk 4]
```diff
@@ -1,144 +1,217 @@
...
+		if(100 to INFINITY)
+			icon_state = "cell-full"
+
+/obj/item/fuel_cell/get_examine_text(mob/user)
+	.  = ..()
+	if(ishuman(user))
+		. += "The fuel indicator reads: [get_fuel_percent()]%"
+
+/obj/item/fuel_cell/proc/get_fuel_percent()
+	return round(100*fuel_amount/max_fuel_amount)
```
Comment (harryob) [inline code/game/machinery/fuelcell_recycler.dm:208]:
```suggestion
	return round(100 * fuel_amount/max_fuel_amount)
```

**Review by harryob — DISMISSED** [review]

**Review by Nanu308 — APPROVED** [review]
Mapping lgtm

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/machinery/fusion_engine.dm [hunk 5]
```diff
@@ -1,460 +1,477 @@
...
+		return
 
-	if(!powernet)
-		if(!connect_to_network())
-			to_chat(user, SPAN_WARNING("Power network not found, make sure the engine is connected to a cable."))
-			return FALSE
+	if(istype(attacking_item, /obj/item/fuel_cell))
+		var/obj/item/fuel_cell/cell = attacking_item
+		if(fusion_cell)
+			to_chat(user, SPAN_WARNING("[src] already has [fusion_cell]. Before you can replace it with [cell] you need to remove it with a wrench."))
```
Comment (Drulikar) [inline code/game/machinery/fusion_engine.dm:274]: It seems this actually wants a crowbar.

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by Drulikar** [comment]
Messages are also really spammy. This is only 1 reactor damaged: 
image

**Review by Drulikar — APPROVED** [review]
As far as I can tell this all is working.

**Pull request merged**